core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -108,6 +113,11 @@ public final void initialize(TelemetryConfiguration configuration) {
         }
     }
 
+    private void initializeMinimumCapabilities(ApplicationInsightsXmlConfiguration userConfiguration, TelemetryConfiguration configuration) {
```

Comment:
Both the channel and the instrumentation key are not capabilities.
Please rename this function to something line `setFundamentalConfiguration`or `setCoreConfiguration`